services/core-api/app/api/verifiable_credentials/manager.py

```diff
@@ -0,0 +1,121 @@
+    # for midware/business level actions between requests and data access
+from time import sleep
+from typing import List
+from celery.utils.log import get_task_logger
+
+from app.tasks.celery import celery
+
+from app.extensions import db
+from app.config import Config
+from app.api.utils.feature_flag import Feature, is_feature_enabled
+
+from app.api.mines.mine.models.mine import Mine
+from app.api.parties.party.models.party import Party
+from app.api.mines.permits.permit.models.permit import Permit
+from app.api.mines.permits.permit_amendment.models.permit_amendment import PermitAmendment
+from app.api.verifiable_credentials.models.credentials import PartyVerifiableCredentialMinesActPermit
+from app.api.verifiable_credentials.models.connection import PartyVerifiableCredentialConnection
+from app.api.services.traction_service import TractionService
+
+task_logger = get_task_logger(__name__)
+
+@celery.task()
+def revoke_all_credentials_for_permit(permit_guid: str, mine_guid:str, reason:str): 
+    cred_exch = PartyVerifiableCredentialMinesActPermit.find_by_permit_guid_and_mine_guid(permit_guid, mine_guid)
+    for ce in cred_exch:
+        traction_svc = TractionService()
+        connection = PartyVerifiableCredentialConnection.find_active_by_party_guid(ce.party_guid)
+        if ce.cred_exch_state in PartyVerifiableCredentialMinesActPermit._active_credential_states:
+            traction_svc.revoke_credential(connection.connection_id, ce.rev_reg_id, ce.cred_rev_id, reason)
+            
+            attempts = 0
+            while not ce.cred_rev_id:
+                sleep(1)
+                db.session.refresh(cred_exch)
+                attempts += 1
+                if attempts > 60:
+                    raise Exception("Never received webhook confirming revokation credential")
```

Comment:
Maybe for the purposes of tracking potential failures it would be simplest to make an update to the cred_exch record to indicate the failure.  Either adding a column to that table, or using the `cred_exch_state`?

Reply:
hmm. true. because if the call to traction returns success but traction fails to successfully revoke, that will probably need different consideration than if that call to traction fails.... hmm. 

Let me take another look. thanks.

Reply:
`error_description` looks good.  Allows for some versatility.